crates/aptos/src/move_tool/mod.rs

```diff
@@ -6,50 +6,122 @@
-use crate::CliResult;
+use crate::{
+    common::{types::MovePackageDir, utils::to_common_result},
+    CliResult, Error,
+};
+use aptos_vm::natives::aptos_natives;
 use clap::{Parser, Subcommand};
-use move_core_types::errmap::ErrorMapping;
-use move_vm_types::gas_schedule::INITIAL_COST_SCHEDULE;
-
-pub mod chain;
+use move_binary_format::CompiledModule;
+use move_cli::package::cli::{run_move_unit_tests, UnitTestResult};
+use move_package::BuildConfig;
+use move_unit_test::UnitTestingConfig;
+use std::{collections::HashSet, path::Path};
 
 /// CLI tool for performing Move tasks
 ///
 #[derive(Subcommand)]
 pub enum MoveTool {
-    Command(MoveCli),
+    Compile(CompilePackage),
+    Test(TestPackage),
 }
 
 impl MoveTool {
     pub async fn execute(self) -> CliResult {
         match self {
-            // TODO: Rethink using the Move CLI and think about how we can make the experience better
-            MoveTool::Command(tool) => tool.execute(),
+            MoveTool::Compile(tool) => to_common_result(tool.execute().await),
+            MoveTool::Test(tool) => to_common_result(tool.execute().await),
         }
     }
 }
 
+/// Compiles a package and returns the [`ModuleId`]s
+#[derive(Parser)]
+pub struct CompilePackage {
+    #[clap(flatten)]
+    move_options: MovePackageDir,
+}
+
+impl CompilePackage {
+    pub async fn execute(&self) -> Result<Vec<String>, Error> {
+        let build_config = BuildConfig {
+            generate_abis: false,
+            generate_docs: true,
+            install_dir: None,
+            ..Default::default()
+        };
+        let compiled_modules = compile_move(
+            build_config,
+            self.move_options.package_dir.as_path(),
+        )?;
+        // TODO: This can be serialized once move is updated
+        let mut ids = Vec::new();
+        compiled_modules
+            .iter()
+            .for_each(|module| ids.push(module.self_id().to_string()));
+        Ok(ids)
+    }
+}
+
+/// Run Move unit tests against a package path
 #[derive(Parser)]
-pub struct MoveCli {
+pub struct TestPackage {
     #[clap(flatten)]
-    move_args: move_cli::Move,
-    #[clap(subcommand)]
-    command: move_cli::Command,
+    move_options: MovePackageDir,
 }
 
-impl MoveCli {
-    fn execute(self) -> CliResult {
-        let error_descriptions: ErrorMapping =
-            bcs::from_bytes(cached_framework_packages::error_map())
-                .map_err(|err| err.to_string())?;
-        move_cli::run_cli(
-            aptos_vm::natives::aptos_natives(),
-            &INITIAL_COST_SCHEDULE,
-            &error_descriptions,
-            &self.move_args,
-            &self.command,
+impl TestPackage {
+    pub async fn execute(&self) -> Result<&'static str, Error> {
+        let config = BuildConfig {
+            test_mode: true,
+            install_dir: Some(tempfile::tempdir().unwrap().path().to_path_buf()),
+            ..Default::default()
+        };
+        // TODO: Use move-package instead of move-cli
```

Comment:
I think this is actually much easier, why not?

Reply:
There's some controls that I'd like to have on automation that allow us to plug into things like a plugin for VScode easier